cmd/kepler/main.go

```diff
@@ -82,8 +65,47 @@ func logVersionInfo(logger *slog.Logger) {
 	)
 }
 
-func waitForInterrupt(ctx context.Context, logger *slog.Logger, signals ...os.Signal) (func() error, func(error)) {
-	ctxInternal, cancel := context.WithCancel(ctx)
+func runServices(logger *slog.Logger, services []service.Service) error {
+	logger.Info("Running all services")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var g run.Group
+	for _, s := range services {
+		runner, ok := s.(service.Runner)
+		if !ok {
+			logger.Warn("skipping service", "service", s.Name())
+			continue
+		}
+
+		g.Add(
+			func() error {
+				logger.Info("Running service", "service", s.Name())
+				return runner.Run(ctx)
+			},
+			func(err error) {
+				if err != nil {
+					logger.Warn("service terminated with error", "service", s.Name(), "error", err)
+				}
+
+				srv, ok := s.(service.Shutdown)
+				if !ok {
+					return
+				}
+
+				logger.Warn("shutting down", "service", s.Name())
+				if shutdownErr := srv.Shutdown(); shutdownErr != nil {
+					logger.Warn("service shutdown failed with error", "service", s.Name(), "error", shutdownErr)
+				}
+			},
+		)
+	}
+	g.Add(waitForInterrupt(ctx, cancel, logger, os.Interrupt))
```

Comment:
add handling for `SIGTERM` . this is used in kubernetes to gracefully shutdown a pod.